app/commands/solution/remove_user_solutions_for_track_from_search_index.rb

```diff
@@ -3,18 +3,27 @@ class Solution::RemoveUserSolutionsForTrackFromSearchIndex
 
   queue_as :background
 
-  initialize_with :user, :track
+  # Use ids here as we're calling this because either the
+  # user or the track has been deleted, so we can't retrieve
+  # the record from the database
+  initialize_with :user_id, :track_id
 
   def call
     Exercism.opensearch_client.delete_by_query(index: Solution::OPENSEARCH_INDEX, body: {
       query: {
         bool: {
           must: [
-            { term: { 'user.id': user.id } },
-            { term: { 'track.id': track.id } }
+            { term: { 'user.id': user_id } },
+            { term: { 'track.id': track_id } }
           ]
         }
       }
     })
+  rescue Elasticsearch::Transport::Transport::Errors::Conflict
+    raise if @retries.to_i >= 3
+
+    sleep(0.5)
+    @retries = @retries.to_i + 1
+    retry
```

Comment:
This might just happen in the tests but if this record is still be persisted internally at cloudsearch when the delete is requested, we get this conflict. I don't see any harm in having this code here.